packages/wdio-cucumber-framework/src/utils.js

```diff
@@ -160,3 +167,57 @@ export function getStepFromFeature(feature, pickle, stepIndex, sourceLocation) {
 
     return targetStep
 }
+
+/**
+ * @param {object[]} result cucumber global result object
+ */
+export function getDataFromResult(result) {
+    return {
+        uri: result[0].uri,
+        feature: result[1],
+        scenarios: result.slice(2)
+    }
+}
+
+export function setUserHookNames (options) {
+    const hooks = [
+        'beforeTestRunHookDefinitions',
+        'beforeTestCaseHookDefinitions',
+        'afterTestCaseHookDefinitions',
+        'afterTestRunHookDefinitions',
+    ]
+    hooks.forEach(hookName => options[hookName].forEach(testRunHookDefinition => {
+        const hookFn = testRunHookDefinition.code
+        if (!hookFn.name.startsWith('wdioHook')) {
+            const userHookBeforeTestRun = (...args) => hookFn(args)
+            testRunHookDefinition.code = userHookBeforeTestRun
+        }
+    }))
+}
+
+export function wrapStepWithHooks (code, config) {
+    const userStepFn = async (...args) => {
```

Comment:
As far as this one is async function I have to either await the result in executeSync or wrap it with future in a similar way to wrapCommand in wdio/sync, but I don't really like the last option.